src/workspaceState.ts

```diff
@@ -9,24 +9,61 @@ import * as vscode from 'vscode';
 import { Disposable } from './common/dispose';
 import { ISessionService } from './services/sessionService';
 import { ILogService } from './services/logService';
-import { filterEvent } from './common/event';
+import { filterEvent, onceEvent } from './common/event';
+
+export type WorkspacePhase = 'unspecified' | 'preparing' | 'imagebuild' | 'pending' | 'creating' | 'initializing' | 'running' | 'interrupted' | 'stopping' | 'stopped';
 
 export class WorkspaceState extends Disposable {
     private workspaceState: WorkspaceStatus | undefined;
+    private _contextUrl: string | undefined;
 
     private _onWorkspaceStateChanged = this._register(new vscode.EventEmitter<void>());
     readonly onWorkspaceStateChanged = this._onWorkspaceStateChanged.event;
 
-    readonly onWorkspaceStopped = filterEvent(this.onWorkspaceStateChanged, () => this.workspaceState?.instance?.status?.phase === WorkspaceInstanceStatus_Phase.STOPPING); // assuming stopping state is never skipped
+    readonly onWorkspaceRunning = onceEvent(filterEvent(this.onWorkspaceStateChanged, () => this.isWorkspaceRunning));
```

Comment:
I will rather have one emitter and fire a number which correspond to order of phase, so client can check `ws.phase > RUNNING`. It will cover stopping and stopped.